src/cmake/axom-config.cmake.in

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) 2017-2020, Lawrence Livermore National Security, LLC and
+# other Axom Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (BSD-3-Clause)
+#------------------------------------------------------------------------------
+
+cmake_minimum_required(VERSION 3.0 FATAL_ERROR)
+
+@PACKAGE_INIT@
+
+if(NOT AXOM_FOUND)
+
+  #----------------------------------------------------------------------------
+  # Set version and paths
+  #----------------------------------------------------------------------------
+  
+  set(AXOM_VERSION       "@AXOM_VERSION_FULL@")
+  set(AXOM_VERSION_MAJOR "@AXOM_VERSION_MAJOR@")
+  set(AXOM_VERSION_MINOR "@AXOM_VERSION_MINOR@")
+  set(AXOM_VERSION_PATCH "@AXOM_VERSION_PATCH@")
+  
+  set(AXOM_INSTALL_PREFIX "@AXOM_INSTALL_PREFIX@")
+  set(AXOM_INCLUDE_DIRS "${AXOM_INSTALL_PREFIX}/include")
+
+  #----------------------------------------------------------------------------
+  # Set user configuration options and features
+  #----------------------------------------------------------------------------
+
+  # Language features
+  set(AXOM_ENABLE_FORTRAN     "@ENABLE_FORTRAN@")
+  set(AXOM_USE_CUDA           "@AXOM_USE_CUDA@")
+  set(AXOM_USE_MPI            "@AXOM_USE_MPI@")
+  set(AXOM_USE_OPENMP         "@AXOM_USE_OPENMP@")
+
+  # Axom components
+  set(AXOM_ENABLE_LUMBERJACK  "@AXOM_ENABLE_LUMBERJACK@")
+  set(AXOM_ENABLE_MINT        "@AXOM_ENABLE_MINT@")
+  set(AXOM_ENABLE_PRIMAL      "@AXOM_ENABLE_PRIMAL@")
+  set(AXOM_ENABLE_QUEST       "@AXOM_ENABLE_QUEST@")
+  set(AXOM_ENABLE_SIDRE       "@AXOM_ENABLE_SIDRE@")
+  set(AXOM_ENABLE_SLAM        "@AXOM_ENABLE_SLAM@")
+  set(AXOM_ENABLE_SLIC        "@AXOM_ENABLE_SLIC@")
+  set(AXOM_ENABLE_SPIN        "@AXOM_ENABLE_SPIN@")
+
+  # Axom built-in TPLs
+  set(AXOM_USE_CLI11          "@AXOM_USE_CLI11@")
+  set(AXOM_USE_FMT            "@AXOM_USE_FMT@")
+  set(AXOM_USE_SPARSEHASH     "@AXOM_USE_SPARSEHASH@")
+
+  # Axom external TPLs
+  set(AXOM_USE_CONDUIT        "@AXOM_USE_CONDUIT@")
+  set(AXOM_USE_HDF5           "@AXOM_USE_HDF5@")
+  set(AXOM_USE_MFEM           "@AXOM_USE_MFEM@")
+  set(AXOM_USE_RAJA           "@AXOM_USE_RAJA@")
+  set(AXOM_USE_SCR            "@AXOM_USE_SCR@")
+  set(AXOM_USE_UMPIRE         "@AXOM_USE_UMPIRE@")
+
+  #----------------------------------------------------------------------------
+  # Bring in required  dependencies for this axom configuration
+  #----------------------------------------------------------------------------
+  include(CMakeFindDependencyMacro)
+
+  # conduit
+  if(AXOM_USE_CONDUIT)
+    set(AXOM_CONDUIT_DIR  "@CONDUIT_DIR@")
+    if(NOT CONDUIT_DIR) 
+      set(CONDUIT_DIR ${AXOM_CONDUIT_DIR}) 
+    endif()
```

Comment:
Is this if-logic here correct? Do we need both `AXOM_CONDUIT_DIR` and `CONDUIT_DIR`?

Couldn't we just do:
```
find_dependency(Conduit REQUIRED NO_DEFAULT_PATHS PATHS "${CONDUIT_DIR}/lib/cmake")
```

Reply:
This logic was an attempt to prioritize a user-provided `FOO_DIR`, 
but if they don't provide it, fall back to axom's exported `AXOM_FOO_DIR`

I didn't want to require the user to specify any of these libraries if they don't want/need to.
For an example, please check out the `using-with-cmake` example, where the user only needs to specify `AXOM_DIR`, but not `CONDUIT_DIR`.

This tracks with what `conduit` and `ascent` were doing.

Reply:
To clarify -- two things are going on here:
1. For each dependency `FOO`, we are importing `AXOM_FOO_DIR` into the user's build system (using a variable from axom's build system)
2. In some (but not all cases), we are also importing the CMake exports of package `FOO`, into the user's build system, using the `find_dependency()` macro.  The path that we're searching for that is either a user-provided `FOO_DIR` or the axom exported `AXOM_FOO_DIR`.

I think part of the confusion relates to where these variables are defined. 
Please note the distinction between the following:
* `@FOO_DIR@` -- this is evaluating axom's variable
* `FOO_DIR` -- this is a string for a variable in the user's build system
* `${FOO_DIR}` -- this is evaluating a variable in the user's build system.

For example, the `CONDUIT_DIR` variable in this file is from the user application's build system, while `@CONDUIT_DIR@` is the path to axom's conduit.

Reply:
I see -- But, `AXOM_<FOO>_DIR` and `<FOO>_DIR` should really be the same?

I am wondering if we should check that and throw an error?

Reply:
I was thinking about that too.
Ideally, they'd be the same, and clearly, problems can arise when they're not, but I'm not sure that it is always a problem if they're different.

Let's leave it as is for now, and fix it down the road if/when problems arise.

Reply:
Some folks move install dirs (to be clear *I don't* recommend this), but this could help mitigate some of the consequences when that is in play.

Reply:
I cannot really think of an instance where someone wants to use a one version of conduit with their application and point to Axom that uses a different version of conduit. I think allowing is bound to cause problems down the road.

Could you please create an issue so that we dont' forget.

Reply:
my example:  it is the same install - but happens to have been relocated to a different place on the file system than that was encoded during a build. 

Folks are doing this now. Turning this into an error could result in many donut payments.
That said - for the know case of this I don't think they are including Axom via cmake

There are other tricky issues with relocating that can't magically be fixed. 
In ascent we have had to provide runtime overrides for folks who choose to move the install dir but still need to access assets (like html templates)

Reply:
> Folks are doing this now. Turning this into an error could result in many donut payments.
> That said - for the know case of this I don't think they are including Axom via cmake

I know exactly what you mean and I totally agree they *shouldn't be moving install directories* 

Also, in this case, Axom, Conduit, Ascent, etc. are not included via CMake.

But, there are instances, where I want to build my own version of Axom for development and point to those "installed" libraries that have been moved. We have been going in and modifying the paths manually in this case. I run into this fairly recently. Having additional guards would be helpful.

Reply:
I would also add that making donut payments for things that folks should not do in the first place encourages the wrong behavior.

So, let's give them an error instead and provide a one-time, generous, donut payment once they *stop* doing things they shouldn't be doing in the first place 😄

Reply:
I recommend *not* advertising the donut reward for changing bad behavior into good. That may encourage certain people to do the wrong thing and then correct themselves just to get a donut.... 🙉

Reply:
@rhornung67 -- too late for that. That's something that has been going on for decades now...I am surprised it is not in the developer's manual (oh wait...you wrote that)

Reply:
See #243. Please, feel free to make additional  comments / suggestions there.

I'll get everyone donuts if we fix this 😄